rtrlib/aspa/aspa.c

```diff
@@ -268,24 +308,41 @@ static enum aspa_status aspa_table_update_internal(struct aspa_table *aspa_table
 			qsort(current->record.provider_asns, current->record.provider_count, sizeof(uint32_t),
 			      compare_asns);
 
-		// $CAS is not stored
-		// Error: Duplicate Remove.
-		if (op_type == ASPA_ADD && existing_record &&
-		    current->record.customer_asn == existing_record->customer_asn) {
+#ifndef ASPA_UPDATE_IN_PLACE
+		struct aspa_record *lastadded_record = aspa_array_get_record(new_array, new_array->size - 1);
+#endif
```

Comment:
I don't think we should access the last record (which by itself I'm against) like that

Reply:
what do you suggest? i used the abstraction here for bounds checks.